src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -616,6 +627,137 @@ private static OperationStatus DecodeWithWhiteSpaceFromUtf8InPlace(Span<byte> ut
+
+            // Non Vbmi path constants
+            Vector512<sbyte> lutHi = Vector512.Create(
+                0x10, 0x10, 0x01, 0x02, 0x04, 0x08, 0x04, 0x08, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10,
+                0x10, 0x10, 0x01, 0x02, 0x04, 0x08, 0x04, 0x08, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10,
+                0x10, 0x10, 0x01, 0x02, 0x04, 0x08, 0x04, 0x08, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10,
+                0x10, 0x10, 0x01, 0x02, 0x04, 0x08, 0x04, 0x08, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10);
+
+            Vector512<sbyte> lutLo = Vector512.Create(
+                0x15, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x13, 0x1A, 0x1B, 0x1B, 0x1B, 0x1A,
+                0x15, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x13, 0x1A, 0x1B, 0x1B, 0x1B, 0x1A,
+                0x15, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x13, 0x1A, 0x1B, 0x1B, 0x1B, 0x1A,
+                0x15, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x13, 0x1A, 0x1B, 0x1B, 0x1B, 0x1A);
+            Vector512<sbyte> mask2F = Vector512.Create((sbyte)'/');
+            Vector512<sbyte> lutShift = Vector512.Create(
+                0, 16, 19, 4, -65, -65, -71, -71, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 16, 19, 4, -65, -65, -71, -71, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 16, 19, 4, -65, -65, -71, -71, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 16, 19, 4, -65, -65, -71, -71, 0, 0, 0, 0, 0, 0, 0, 0);
+            Vector512<sbyte> packBytesInLaneMask = Vector512.Create(
+                        2, 1, 0, 6, 5, 4, 10, 9, 8, 14, 13, 12, -1, -1, -1, -1,
+                        18, 1, 0, 6, 5, 4, 10, 9, 8, 14, 13, 12, -1, -1, -1, -1,
+                        2, 1, 0, 6, 5, 4, 10, 9, 8, 14, 13, 12, -1, -1, -1, -1,
+                        2, 1, 0, 6, 5, 4, 10, 9, 8, 14, 13, 12, -1, -1, -1, -1);
+            Vector512<int> packLanesControl = Vector512.Create(0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14, 0, 0, 0, 0);
+
+            // Common path constants
+            Vector512<sbyte> mergeConstant0 = Vector512.Create(0x01400140).AsSByte();
+            Vector512<short> mergeConstant1 = Vector512.Create(0x00011000).AsInt16();
+
+            // The fastest version of this algorithm requires AVX512VBMI support.
+            // The fallback path uses AVX512BW and AVX512F instructions.
+            do
+            {
+                AssertRead<Vector512<sbyte>>(src, srcStart, sourceLength);
+                Vector512<sbyte> str = Vector512.Load(src).AsSByte();
+
+                // Step 1: Translate encoded Base64 input to their original indices
+                // This step also checks for invalid inputs and exits.
+                // After this, we have indices which are verified to have upper 2 bits set to 0 in each byte.
+                // origIndex      = [...|00dddddd|00cccccc|00bbbbbb|00aaaaaa]
+                Vector512<sbyte> origIndex;
+                if (Avx512Vbmi.IsSupported)
```

Comment:
Do we want to duplicate code for missing Avx512 subsets that much? Can we just guard the whole impl to require `Avx512Vbmi` and `Avx512BW` I doubt Base64's perf and large inputs are that important, IMO.

Reply:
this also complicates test matrix

Reply:
This is a good question and I'm open to it. Would like to hear more input before makinga  call @tannergooding - thoughts?

Reply:
I modified the commit to have just the VBMI path after a conversation with Tanner